app/controllers/bills_controller.rb

```diff
@@ -25,19 +30,35 @@ def create
 
   def update
     @bill = Bill.find params[:id]
-    if @bill.cancelled!
-      flash[:info] = t "global.cancel_success"
+    if @bill.update bill_params
+      flash[:success] = t "global.update_success"
+      redirect_to user_path current_user.id
     else
-      flash[:danger] = t "global.cancel_error"
+      flash.now[:danger] = t "global.upate_service_error"
+      redirect_to bookings_path bill_id: params[:id]
     end
-    @bills = current_user.bills.waiting.bill_created_at.page(params[:page])
-                         .per Settings.bill.page
-    :js
   end
 
   private
+  def day_between checkin, checkout
+    (checkout - checkin) / Settings.mini + Settings.tm
+  end
+
+  def price_booking checkin, checkout, price
+    price * day_between(checkin, checkout)
+  end
+
+  def update_booking_and_bill
+    total_price = 0
+    @bill.bookings.each do |book|
+      price = price_booking(book.checkin, book.checkout, book.room.price).to_i
+      book.update price: price, status: 0
+      total_price += price
+    end
+    @bill.update price: total_price
```

Comment:
ai lại làm thế e
viết hàm tính toán total_price
xong phần params @bill.update bill_params.merge(price: total_price)
@bill chỉ cần update 1 lần thôi

Reply:
@Trancuc 

> viết hàm tính toán total_price
> xong phần params @bill.update bill_params.merge(price: total_price)